modules/tinymce/src/core/main/ts/api/Options.ts

```diff
@@ -437,6 +438,23 @@ const register = (editor: Editor): void => {
     default: false
   });
 
+  registerOption('disabled', {
+    processor: (value) => {
+      if (Type.isBoolean(value)) {
+        if (editor.initialized && isDisabled(editor) !== value) {
+          // Schedules the callback to run in the next microtask queue once the option is updated
+          Promise.resolve().then(() => {
+            fireDisabledStateChange(editor, value);
+          });
+        }
+        return { valid: true, value };
+      }
+
+      return { valid: false, message: 'The value must be a boolean.' };
+    },
+    default: false
+  });
```

Comment:
Yep, it's possible to use `options.set('disabled', true/false)` at runtime to enable or disable the editor. The UI elements and editor properties will update automatically. The processor is called every time this option is set.

Reply:
Would this be better exposed as an API instead? We don't generally expect initialization options to be updated

Reply:
Yea, I see your point. In this case, the dynamic `options.set()` gives us simpler way of handling editor state changes without needing separate enable/disable APIs (for future options as well). I'm happy to discuss if you see advantages to the API approach.

Reply:
I'm happy for us to handle it this way for this release. I don't think we should document that this can be changed dynamically, so if we get requests for it we can discuss the best way to expose it then.

Reply:
Having it be dynamic is what made this PR difficult, a customer confirmed they wanted exactly what `editor.mode.set('readonly')` used to do including runtime toggle. I decided I didn't want a dedicated API for it.

Yes it's not something we have done in the past, but we've had a few requests for this sort of dynamic options behaviour. Features that call `editor.options.get()` every time they are used are already dynamic.

While some of these "locked at load time" options are extremely difficult to make dynamic, it would be nice if we could look at ways to make this easy. Perhaps we'll end up supporting a post-processor in option registration, I'd prefer to avoid a wide-scale "option changed" event.

Reply:
The mode api is the thing that should be used for flipping between UI states like this. So not sure why this couldn't be done with flipping readonly on/off and having a additional option that just controls what the readonly mode does like readonly_disabled_ui or something that is enabled by default and that the customer can disable that one doesn't need to be dynamic.

Reply:
This whole thing started that way, along these lines:

```
editor.mode.set({
  readonly_ui: true,
  readonly_content: true
});
```

But it seemed overly complex, both to implement and explain.

Reply:
Right so @spocke's suggestion is to add something on top of the readonly mode, that controls that whether we should disable everything or not. That way we don't need new events nor dynamically switching the option at run time and can reuse the `SwitchMode` as well since we have the modes API for that.